### PR TITLE
Support random uid

### PR DESCRIPTION
**Pull request by csrwng**
Initial attempt at supporting non-root user

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal ... it's my naive first shot at running as non-root

File: 5.5/Dockerfile [hunk 1]
```diff
@@ -25,6 +25,12 @@ RUN yum install -y \
     yum clean all -y && \
     mkdir /tmp/sessions
 
+RUN chmod -R og+rwx /opt/rh/php55/root/etc && \
+    chmod -R og+rwx /opt/rh/httpd24/root/var/run/httpd && \
+    chmod -R og+rwx /opt/rh/httpd24/root/etc/httpd/logs && \
+    cat /opt/rh/httpd24/root/etc/httpd/conf/httpd.conf | sed "s/^Listen 80/#Listen 80/" > /tmp/httpd.conf && \
```
Comment (bparees) [inline 5.5/Dockerfile]: sed -i

and since you're commenting out listen on 80, what's it listening on?

**Comment by bparees** [comment]
does this work as is?  @jhadvig indicated there were problems writing the PID file, but maybe it's writing it to one of the paths you chmod'd.

File: 5.5/Dockerfile (hunk 1 above)
Comment (csrwng) [inline 5.5/Dockerfile]: 8080 ... it's in the httpd.conf we copy over

**Comment by csrwng** [comment]
it does work as is... however not ideal... because the sti build still happens as root. However, the resulting image can be run as any user

**Comment by csrwng** [comment]
@bparees - ready for a more serious review. Made the change to rhel image and updated tests to try with a random uid. Requires https://github.com/openshift/source-to-image/pull/177

File: 5.5/Dockerfile [hunk 2]
```diff
@@ -25,9 +25,18 @@ RUN yum install -y \
     yum clean all -y && \
     mkdir /tmp/sessions
 
-ENV PATH $PATH:/opt/rh/php55/root/usr/bin
+RUN sed -i -e 's/^Listen 80/Listen 0.0.0.0:8080/' \
```
Comment (bparees) [inline 5.5/Dockerfile]: I wonder if it's worth putting this in an external script that we ADD and RUN.  what do you think?  (you can use the contrib dir I believe)

**Comment by csrwng** [comment]
@bparees after looking at the sti-perl repo, I think a better approach is to simply comment out what we don't need from the base httpd.conf and merge the one in the contrib directory. That way I can be more consistent with both projects.

File: 5.5/Dockerfile (hunk 2 above)
Comment (csrwng) [inline 5.5/Dockerfile]: yup, makes sense

**Comment by csrwng** [comment]
updated to use an external .sed file for common httpd.conf replacements

**Comment by csrwng** [comment]
[test]

File: 5.5/contrib/httpdconf.sed [hunk 3]
```diff
@@ -0,0 +1,7 @@
+s/^Listen 80/Listen 0.0.0.0:8080/
+s/^User apache/User default/
+s/^Group apache/Group default/
+s%^DocumentRoot "/opt/rh/httpd24/root/var/www/html"%DocumentRoot "/opt/openshift/src"%
+s%^<Directory "/opt/rh/httpd24/root/var/html"%<Directory "/opt/openshift/src"%
+s%^ErrorLog "logs/error_log"%ErrorLog "/tmp/error_log"%
+s%CustomLog "logs/access_log"%CustomLog "/tmp/access_log"%
```
Comment (bparees) [inline 5.5/contrib/httpdconf.sed]: can we bounce these logs to STDOUT instead?

Comment (bparees) [inline 5.5/contrib/httpdconf.sed]: actually i guess let's not, for now..but we probably have to decide what to do about it at some point.

**Comment by bparees** [comment]
lgtm but would like to sort out why it seems to be failing the sti-php test build (possibly just the bad STI version issue you mentioned)

**Comment by bparees** [comment]
rerunning the test here:
https://ci.openshift.redhat.com/jenkins/view/Image%20Verification/job/sti-php/10/

hopefully that picks up the new AMI that should have your STI changes.

**Pull request merged**